packages/flutter/test/material/localizations_test.dart

```diff
@@ -293,4 +293,31 @@ void main() {
     expect(tester.widget<Text>(find.byKey(textKey)).data, 'id_JV');
   });
 
+  testWidgets('Localizations is compatible with ChangeNotifier.dispose() called during didChangeDependencies', (WidgetTester tester) async {
+    await tester.pumpWidget(new MaterialApp(
+      supportedLocales: const <Locale>[
+        const Locale('en', 'US'),
+        const Locale('es', 'ES'),
+      ],
+      localizationsDelegates: <_DummyLocalizationsDelegate>[
+        new _DummyLocalizationsDelegate(),
+      ],
+      // PageView calls ScrollPosition.dispose() during didChangeDependencies.
+      home: new PageView(),
+    ));
+
+    await tester.binding.setLocale('es', 'US');
+    await tester.pump();
+    await tester.pumpWidget(new Container());
+  });
+}
+
+class _DummyLocalizationsDelegate extends LocalizationsDelegate<DummyLocalizations> {
```

Comment:
All of the other classes that the tests depend on are at the top of the file. Why are these at the bottom? A name that was a little more telling than "Dummy" might be useful.

Reply:
Moved up and added a doc comment explaining why this dummy is needed. It's "dummy" because it does not contain any localization data.